### PR TITLE
Add Azure Monitor Distributions

### PR DESCRIPTION
**Pull request by mattmccleary**
Language Distros for .NET, Java, Node.js, and Python

### CODE REVIEW

File: data/ecosystem/distributions.yaml [hunk 1]
```diff
@@ -83,3 +83,19 @@
   url: https://github.com/signalfx/splunk-otel-go
   docsUrl: https://docs.splunk.com/observability/en/gdi/get-data-in/application/go/get-started.html
   components: [Go]
+- name: Azure Monitor Distribution of OpenTelemetry ASP.NET Core
+  url: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore
+  docsUrl: https://learn.microsoft.com/azure/azure-monitor/app/opentelemetry-enable
+  components: [.NET]
+- name: Azure Monitor Distribution of OpenTelemetry Java
+  url: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore
```
Comment (rajkumar-rangaraj) [inline data/ecosystem/distributions.yaml:91]: This should point to Java repository.

Comment (mattmccleary) [reply data/ecosystem/distributions.yaml]: Good catch. Updated link.

File: data/ecosystem/distributions.yaml [hunk 2]
```diff
@@ -83,3 +83,19 @@
   url: https://github.com/signalfx/splunk-otel-go
   docsUrl: https://docs.splunk.com/observability/en/gdi/get-data-in/application/go/get-started.html
   components: [Go]
+- name: Azure Monitor Distribution of OpenTelemetry ASP.NET Core
+  url: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore
+  docsUrl: https://learn.microsoft.com/azure/azure-monitor/app/opentelemetry-enable
```
Comment (cijothomas) [inline data/ecosystem/distributions.yaml:88]:
```suggestion
  docsUrl: https://learn.microsoft.com/azure/azure-monitor/app/opentelemetry-enable?tabs=aspnetcore
```

Could we put more specific links?

Comment (mattmccleary) [reply data/ecosystem/distributions.yaml]: Updated.

**Review by cijothomas — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Pull request merged**